src/extensions/nginx-app-protect/monitoring/processor/nap.go

```diff
@@ -430,12 +437,25 @@ func (f *NAPConfig) getViolations(logger *logrus.Entry) []*models.ViolationData
 
 			if v.Header != (Header{}) {
 				isB64Decoded = v.Header.IsBase64Decoded
-				name = v.Header.Name
-				value = v.Header.Value
+				if v.Header.Name != "" || v.Header.Value != "" {
+					name = v.Header.Name
+					value = v.Header.Value
+				} else {
+					value = v.Header.Text
+				}
 			} else if v.HeaderData != (Header{}) {
 				isB64Decoded = v.HeaderData.IsBase64Decoded
 				name = v.HeaderData.Name
 				value = v.HeaderData.Value
+			} else if v.HeaderLength != "" {
+				isB64Decoded = true
```

Comment:
question: is this a safe assumption to make that whenever we have HeaderLength, the Header name is B64 encoded?

Reply:
Based on all the examples we came across the assumption is correct.